src/snovault/viewconfigs/base_view.py

```diff
@@ -44,6 +52,34 @@ def __init__(self, context, request):
         from_, page_size = get_pagination(request)
         self._from_ = from_
         self._size = page_size
+        self._view_name = '' # should be set by view
+        self._factory_name = ''  # view should set this
+
+
+    def _validate_items(self, type_info=None):
+        """
+        Ensure query string parameters are valid and throws a custom HTTP-500 error if not.
+
+            :param type_info=None: type information
+            :error: Custom HTTP-500 if query string parameters are invalid
+        """
+        msg = None
+        if len(self._doc_types) != 1:
+            msg = (
+                'Search result {} currently requires specifying a '
+                'single type.'.format(
+                    self._view_name,
+                )
+            )
+        elif self._doc_types[0] not in self._types:
+            msg = 'Invalid type: {}'.format(self._doc_types[0])
+        elif type_info and self._factory_name and not hasattr(type_info.factory, self._factory_name):
```

Comment:
I could not think of a better way... Some code needs to validate type_info and some parts do not. Some parts need _factory_name and some do not. But they share a lot of common validation needs.